CONTRIBUTING.md

```diff
@@ -179,6 +179,11 @@ Ok(match foo {
 })
 ```
 
+#### Avoid `ref` in match patterns
+
+When writing match expressions, try to avoid using `ref` in patterns. Prefer
+taking a reference on the subject of the `match`.
```

Comment:
The "subject" is technically called the `scrutinee`. Maybe refer to the Rust reference for readers who are unfamiliar with that expression?

The feature was introduced relatively late and is known as "binding modes", maybe worth referencing https://rust-lang.github.io/rfcs/2005-match-ergonomics.html or do you think that's too detailed?

Also consider adding some reasoning, like that it is unidiomatic in today's Rust and that the meaning of the `ref` keywords is a little unfamiliar since it is used here to take a reference (unlike the use of `&` that's usually used in expression contexts).

Reply:
> The "subject" is technically called the scrutinee. Maybe refer to the Rust reference for readers who are unfamiliar with that expression?

Sounds good. I wasn't sure if `scrutinee` was well know but with a link to the reference I think it's fine :+1: 

> Also consider adding some reasoning

Added some more reasoning and a link to the match ergonomics RFC. I think it's fair to link to more detailed resources.